compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -45,6 +45,7 @@ rustc_index::newtype_index! {
 
 impl DepNodeIndex {
     pub const INVALID: DepNodeIndex = DepNodeIndex::MAX;
+    pub const DUMMY_ANON: DepNodeIndex = DepNodeIndex::from_u32(0);
```

Comment:
Let's rename this to something more descriptive like `SINGLETON_DEPENDENCYLESS_ANON_NODE`.